liquidation-monitor/src/agent.js

```diff
@@ -0,0 +1,389 @@
+    if (data.supply[tokenAddress] === undefined) data.supply[tokenAddress] = {};
+  }
+}
+
+// Initializes data required for handler
+function provideInitialize(data) {
+  return async function initialize() {
+    // Assign configurable fields
+    data.protocolName = config.protocolName;
+    data.protocolAbbreviation = config.protocolAbbreviation;
+    data.developerAbbreviation = config.developerAbbreviation;
+    data.alert = config.liquidationMonitor.alert;
+    data.minimumLiquidationInUSD = config.liquidationMonitor.triggerLevels.minimumLiquidationInUSD;
+    data.lowHealthThreshold = config.liquidationMonitor.triggerLevels.lowHealthThreshold;
+    data.cTokenABI = getAbi('cErc20.json');
+    data.provider = getEthersProvider();
+    data.accounts = {}; // Health of all accounts, calcHealth, lastUpdated, [assetsIn addresses]?
+    data.supply = {}; // qty of cTokens (not Tokens)
+    data.borrow = {}; // qty of Tokens (not cTokens)
+    data.tokens = {}; // each cToken's address, symbol, contract, ratio, price, lastUpdatePrice
+    data.newAccounts = []; // New account from transaction events
+    data.totalNewAccounts = 0;
+
+    // Compound API filter and Comptroller contract
+    const {
+      comptrollerAddress,
+      oneInchAddress,
+      triggerLevels: {
+        maximumHealth,
+        minimumBorrowInETH,
+      },
+    } = config.liquidationMonitor;
+    const comptrollerABI = getAbi(config.liquidationMonitor.comptrollerABI);
+    const oneInchABI = getAbi(config.liquidationMonitor.oneInchABI);
+    data.comptrollerAddress = comptrollerAddress;
+    data.comptrollerContract = new ethers.Contract(
+      comptrollerAddress,
+      comptrollerABI,
+      data.provider,
+    );
+    data.oneInchContract = new ethers.Contract(
+      oneInchAddress,
+      oneInchABI,
+      data.provider,
+    );
+
+    // Get initial accounts from Compound API
+    // Find total number of results with the first request
+    const initialRequest = buildJsonRequest(maximumHealth, minimumBorrowInETH, 1, 1);
+    const initialResults = await callCompoundAPI(initialRequest);
+    const totalEntries = initialResults.pagination_summary.total_entries;
+
+    // Determine number of pages needed to query.
+    const pageIncrement = 100;
+    const maxPages = Math.ceil(totalEntries / pageIncrement);
+
+    // Query each page and add accounts. Starting at 1 and including maxPages
+    const foundAccounts = [];
+    // Shorthand Range() function. ( Ex: 5 => [1,2,3,4,5] )
+    const pages = [...Array(maxPages)].map((_, i) => 1 + i);
+    await Promise.all(pages.map(async (page) => {
+      const currentRequest = buildJsonRequest(
+        maximumHealth, minimumBorrowInETH, page, pageIncrement,
+      );
+      const apiResults = await callCompoundAPI(currentRequest);
+      apiResults.accounts.forEach((currentAccount) => {
+        foundAccounts.push(currentAccount);
+      });
+    }));
```

Comment:
Instead of pushing the individual elements of the `apiResults.accounts` Array into the `foundAccounts` Array, you should be able to wait for all of the promises to settle, which should return an Array of Arrays, then use `.flat()` to create `foundAccounts`.
```suggestion
    // Query each page and add accounts. Starting at 1 and including maxPages
    // Shorthand Range() function. ( Ex: 5 => [1,2,3,4,5] )
    const pages = [...Array(maxPages)].map((_, i) => 1 + i);
    const foundAccounts = await Promise.all(pages.map(async (page) => {
      const currentRequest = buildJsonRequest(
        maximumHealth, minimumBorrowInETH, page, pageIncrement,
      );
      const apiResults = await callCompoundAPI(currentRequest);
      return apiResults.accounts;
    })).flat();
```

Reply:
Implemented in a5e28621d461abcaac565530d37c2979cd9518ba